apps/web/src/ui/swap/twap/twap.tsx

```diff
@@ -0,0 +1,305 @@
+import { ChainId } from 'sushi/chain'
+import { Currency } from 'sushi/currency'
+import { useAccount, useChainId, useSwitchChain } from 'wagmi'
+
+import { TokenSelector } from 'src/lib/wagmi/components/token-selector/TokenSelector'
+import { Checker } from 'src/lib/wagmi/systems/Checker'
+import { Address } from 'viem'
+import {
+  useDerivedStateSimpleSwap,
+  useSimpleSwapTrade,
+} from '../simple/derivedstate-simple-swap-provider'
+import { SimpleSwapBridgeBanner } from '../simple/simple-swap-bridge-banner'
+import { SimpleSwapHeader } from '../simple/simple-swap-header'
+import { SimpleSwapSettingsOverlay } from '../simple/simple-swap-settings-overlay'
+import { SwapModeButtons } from '../swap-mode-buttons'
+import { DCAMaintenanceMessage } from './dca-maintenance-message'
+import { LimitMaintenanceMessage } from './limit-maintenance-message'
+import { useIsDCAMaintenance } from './use-is-dca-maintenance'
+import { useIsLimitMaintenance } from './use-is-limit-maintenance'
+
+const Modal = ({
+  open,
+  onClose,
+  title,
+  children,
+  header,
+}: {
+  open: boolean
+  onClose: () => void
+  title?: string
+  children?: ReactNode
+  header?: ReactNode
+}) => {
+  return (
+    <Dialog open={open} onOpenChange={onClose}>
+      <DialogContent>
+        <DialogHeader>
+          {header}
+          {title && <DialogTitle>{title}</DialogTitle>}
+        </DialogHeader>
+        {children}
+      </DialogContent>
+    </Dialog>
+  )
+}
+
+const TokenSelectModal = ({
+  selected,
+  children,
+  onSelect,
+}: {
+  selected: Currency
+  children: ReactNode
+  onSelect: (currency: Currency) => void
+}) => {
+  const {
+    state: { chainId },
+  } = useDerivedStateSimpleSwap()
+
+  return (
+    <TokenSelector
+      id={`twap-token-selector`}
+      selected={selected}
+      chainId={chainId}
+      onSelect={onSelect}
+      includeNative={true}
+      hidePinnedTokens={false}
+      hideSearch={false}
+    >
+      {children}
+    </TokenSelector>
+  )
+}
+
+const useTrade = () => {
+  const { data: trade, isInitialLoading } = useSimpleSwapTrade()
+
+  return {
+    isLoading: isInitialLoading,
+    outAmount: trade?.amountOut?.quotient.toString(),
+  }
+}
+
+const usePriceUSD = (address?: Address) => {
+  const {
+    state: { chainId },
+  } = useDerivedStateSimpleSwap()
+
+  const { data: price } = usePrice({
+    chainId,
+    enabled: true,
+    address,
+  })
+
+  return price?.toSignificant(6)
+}
+
+const getTokenLogo = (currency: Currency) => {
+  try {
+    const src = currency.isNative
+      ? `native-currency/ethereum.svg`
+      : `tokens/${currency.chainId}/${currency.wrapped.address}.jpg`
+    const params = ['f_auto', 'c_limit', `w_${64}`]
+    return `https://cdn.sushi.com/image/upload/${params.join(
+      ',',
+    )}/d_unknown.png/${src}`
+  } catch (_error) {
+    return ''
+  }
+}
+
+const useTokenList = () => {
+  const { data: customTokenMap } = useCustomTokens()
```

Comment:
right when we're killing local token lists... :/

Reply:
is what it is, need to replace across the rest of the app anyway